src/drawing/__tests__/DrawingAnnotations-test.tsx

```diff
@@ -166,5 +212,60 @@ describe('DrawingAnnotations', () => {
 
             expect(wrapper.find(PopupDrawingToolbar).hasClass('ba-is-drawing')).toBe(true);
         });
+
+        describe('toolbar', () => {
+            const getWrapperWithToolbar = (
+                drawnPathGroups: Array<PathGroup>,
+                stashedPathGroups: Array<PathGroup>,
+            ): ReactWrapper =>
+                getWrapper({
+                    canShowPopupToolbar: true,
+                    isCreating: true,
+                    drawnPathGroups,
+                    stashedPathGroups,
+                });
+
+            test.each`
+                drawn         | stashed       | canComment
+                ${pathGroups} | ${[]}         | ${true}
+                ${[]}         | ${pathGroups} | ${false}
+                ${pathGroups} | ${pathGroups} | ${true}
+            `(
+                'should set canComment to $canComment if drawnPathGroups is $drawn and stashedPathGroups is $stashed',
+                ({ drawn, stashed, canComment }) => {
+                    const wrapper = getWrapperWithToolbar(drawn, stashed);
+
+                    expect(wrapper.find(PopupDrawingToolbar).prop('canComment')).toBe(canComment);
+                },
+            );
+
+            test.each`
+                drawn         | stashed       | canRedo
+                ${pathGroups} | ${[]}         | ${false}
+                ${[]}         | ${pathGroups} | ${true}
+                ${pathGroups} | ${pathGroups} | ${true}
+            `(
+                'should set canRedo to $canRedo if drawnPathGroups is $drawn and stashedPathGroups is $stashed',
+                ({ drawn, stashed, canRedo }) => {
+                    const wrapper = getWrapperWithToolbar(drawn, stashed);
+
+                    expect(wrapper.find(PopupDrawingToolbar).prop('canRedo')).toBe(canRedo);
+                },
+            );
+
+            test.each`
+                drawn         | stashed       | canUndo
+                ${pathGroups} | ${[]}         | ${true}
+                ${[]}         | ${pathGroups} | ${false}
+                ${pathGroups} | ${pathGroups} | ${true}
+            `(
+                'should set canUndo to $canUndo if drawnPathGroups is $drawn and stashedPathGroups is $stashed',
+                ({ drawn, stashed, canUndo }) => {
+                    const wrapper = getWrapperWithToolbar(drawn, stashed);
+
+                    expect(wrapper.find(PopupDrawingToolbar).prop('canUndo')).toBe(canUndo);
```

Comment:
Seems like these three `test.each` can be combined into one with a couple additional columns for each `can` check.